src/Tasks.UnitTests/XamlTaskFactory_Tests.cs

```diff
@@ -191,6 +191,30 @@ public void TestBasicReversibleBooleanSwitches()
             Assert.Equal(PropertyType.Boolean, properties.First.Value.Type);
         }
 
+        [Fact]
+        public void TestParseIncorrect_PropertyNamesMustBeUnique()
+        {
+            bool exceptionCaught = false;
+
+            try
+            {
+                string incorrectXmlContents = @"<ProjectSchemaDefinitions xmlns=`clr-namespace:Microsoft.Build.Framework.XamlTypes;assembly=Microsoft.Build.Framework` xmlns:x=`http://schemas.microsoft.com/winfx/2006/xaml` xmlns:sys=`clr-namespace:System;assembly=mscorlib` xmlns:impl=`clr-namespace:Microsoft.VisualStudio.Project.Contracts.Implementation;assembly=Microsoft.VisualStudio.Project.Contracts.Implementation`>
+                                     <Rule Name=`CL`>
+                                       <BoolProperty Name=`SameName` Switch=`Og` ReverseSwitch=`Og-` />
+                                       <BoolProperty Name=`SameName` Switch=`Og` ReverseSwitch=`Og-` />
+                                     </Rule>
+                                   </ProjectSchemaDefinitions>";
+                TaskParser tp = XamlTestHelpers.LoadAndParse(incorrectXmlContents, "CL");
+            }
+            catch (XamlParseException ex)
+            {
+                exceptionCaught = true;
+                Assert.Equal("MSB3724: Unable to create Xaml task.  Duplicate property name 'SameName'.", ex.Message);
+            }
+
+            Assert.True(exceptionCaught); // "Should have caught a XamlParseException"
```

Comment:
We're trying to use Shouldly where we can (for debugging purposes).
```suggestion
            exceptionCaught.ShouldBeTrue(); // "Should have caught a XamlParseException"
```